frame/support/src/traits/storage.rs

```diff
@@ -90,3 +92,35 @@ impl StorageInfoTrait for Tuple {
 pub trait PartialStorageInfoTrait {
 	fn partial_storage_info() -> Vec<StorageInfo>;
 }
+
+/// Allows a pallet to specify storage keys to whitelist during benchmarking.
+/// This means those keys will be excluded from the benchmarking performance
+/// calculation.
+pub trait WhitelistedStorageKeys {
+	/// Returns a [`Vec<TrackedStorageKey>`] indicating the storage keys that
+	/// should be whitelisted during benchmarking. This means that those keys
+	/// will be excluded from the benchmarking performance calculation.
+	fn whitelisted_storage_keys() -> Vec<TrackedStorageKey>;
+}
+
+#[cfg_attr(all(not(feature = "tuples-96"), not(feature = "tuples-128")), impl_for_tuples(64))]
+#[cfg_attr(all(feature = "tuples-96", not(feature = "tuples-128")), impl_for_tuples(96))]
+#[cfg_attr(feature = "tuples-128", impl_for_tuples(128))]
+impl WhitelistedStorageKeys for Tuple {
+	fn whitelisted_storage_keys() -> Vec<TrackedStorageKey> {
+		// de-duplicate the storage keys
+		// use BTreeMap so the resulting list of keys is unique
+		let mut combined_keys: BTreeMap<Vec<u8>, TrackedStorageKey> = BTreeMap::new();
+		for_tuples!( #(
+			for storage_key in Tuple::whitelisted_storage_keys() {
+				combined_keys.insert(storage_key.key.clone(), storage_key);
+			}
+		 )* );
+		// flatten BTreeMap down to a vec
+		let mut final_combined_keys: Vec<TrackedStorageKey> = Vec::new();
+		for storage_key in combined_keys.values() {
+			final_combined_keys.push(storage_key.clone());
+		}
```

Comment:
```suggestion
		combined_keys.values().collect::<Vec<_>>()
```

Reply:
Just a note had to change to the following because as written you'll get a `Vec<&TrackedStorageKey>` instead of a `Vec<TrackedStorageKey>`, and there is an ownership issue requiring the `clone()` instead of just de-referencing. If anyone knows a simpler way of writing this I'm all ears!

```rust
combined_keys.values().map(|e| e.clone()).collect::<Vec<_>>()
```

Reply:
https://paritytech.github.io/substrate/master/sp_std/iter/struct.Cloned.html

Reply:
Ah no you know what it was it's this:
```
the trait `core::cmp::Ord` is not implemented for `TrackedStorageKey`
```

Reply:
done! thanks!